tests/main/xdg-runtime-dir/task.yaml

```diff
@@ -0,0 +1,16 @@
+summary: Ensure that XDG_RUNTIME_DIR is created
+
+details: |
+    Ensure that the XDG_RUNTIME_DIR is created when a snap is launched.
+
+prepare: |
+    echo Testing XDG_RUNTIME_DIR
+    snap install --edge test-snapd-xdg-dir
+    rm -f /run/user/0/snap.test-snapd-xdg-dir
+
+execute: |
+    test ! -e /run/user/0/snap.test-snapd-xdg-dir
+    #   xdgtest returns 255 if the folder doesn't exist,
+    #   making the task fail in that case.
+    snap run test-snapd-xdg-dir || echo "The XDG_RUNTIME_DIR folder doesn't exist inside the environment"
+    test -d /run/user/0/snap.test-snapd-xdg-dir
```

Comment:
Consider adding a second call to the  test snap to show that things work if the directory already exists (i.e. that we're correctly handling the error from mkdir()).

Reply:
Good point! Done.